src/SimSpace/SimFormat.hs

```diff
@@ -337,13 +338,18 @@ maybeShowVersionAndExit = do
     putStrLn $ showVersion Paths.version
     exitSuccess
 
-main :: IO ()
-main = do
-  maybeShowVersionAndExit
-  (nonimports, importsAndAfter) <- break ("import " `isPrefixOf`) . lines <$> getContents
-  traverse_ outputBlock $ reassemble nonimports $ untilNothing processBlock (chunkedInputs importsAndAfter)
+reformat :: [Line] -> [Line]
+reformat programLines =
+  let
+    (nonimports, importsAndAfter) = break ("import " `isPrefixOf`) programLines
+  in
+    concatMap blockToLines . reassemble nonimports $ untilNothing processBlock (chunkedInputs importsAndAfter)
 
   where
+    blockToLines :: Block -> [String]
+    blockToLines (Left s) = lines (s <> "\n")
+    blockToLines (Right s) = lines s
```

Comment:
`lines "foo\n"` and `lines "foo"` both result in `["foo"]`. How about `lines s ++ [""]` instead of `lines (s <> "\n")`?